Flaky Test: Failed to build images at main branch

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
Building images at the main branch is always failing after we merge https://github.com/kubernetes-sigs/kueue/pull/8369

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/post-kueue-push-images/2002021275274317824

```shell
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20250303144028-a0af3efb3deb
DEST_CHART_DIR= \
HELM="/workspace/bin/helm" YQ="/workspace/bin/yq" GIT_TAG="v0.16.0-devel-151-g4722f21b" IMAGE_REGISTRY="us-central1-docker.pkg.dev/k8s-staging-images/kueue" \
HELM_CHART_PUSH=true \
./hack/helm-chart-package.sh
rm: can't remove 'cmd/experimental/kueue-populator/charts/kueue-populator/charts': No such file or directory
make: *** [Makefile:319: helm-chart-package] Error 1
ERROR
ERROR: build step 0 "gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20250513-9264efb079" failed: step exited with non-zero status: 2
--------------------------------------------------------------------------------
```

**What you expected to happen**:
No errors

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/hack/helm-chart-package.sh
+++ b/hack/helm-chart-package.sh
@@ -76,8 +76,8 @@ ${YQ}  e  ".kueueViz.frontend.image.repository = \"${IMAGE_REGISTRY}/kueueviz-fr
 ${YQ}  e  ".kueuePopulator.image.repository = \"${IMAGE_REGISTRY}/kueue-populator\" | .kueuePopulator.image.tag = \"${GIT_TAG}\" | .kueuePopulator.image.pullPolicy = \"IfNotPresent\"" -i cmd/experimental/kueue-populator/charts/kueue-populator/values.yaml
 
 # TODO: consider signing it
-rm -R cmd/experimental/kueue-populator/charts/kueue-populator/charts
-rm cmd/experimental/kueue-populator/charts/kueue-populator/Chart.lock
+rm -Rf cmd/experimental/kueue-populator/charts/kueue-populator/charts
+rm -f cmd/experimental/kueue-populator/charts/kueue-populator/Chart.lock
 ${HELM} package --version "${chart_version}" --app-version "${GIT_TAG}" charts/kueue -d "${DEST_CHART_DIR}"
 ${HELM} dependency build cmd/experimental/kueue-populator/charts/kueue-populator
 ${HELM} package --version "${chart_version}" --app-version "${GIT_TAG}" cmd/experimental/kueue-populator/charts/kueue-populator -d "${DEST_CHART_DIR}"

